File: tests/cases/with_open.py
#!/usr/bin/env python3

import os
from tempfile import NamedTemporaryFile


if __name__ == "__main__":
    with NamedTemporaryFile(mode="a+", delete=False) as temp_file:
        file_path = temp_file.name
        with open(file_path, "w") as f:
            f.write("hello")
        with open(file_path, "r") as f:
            print(f.read())

File: pyjl/tests/tests_py2many-py2many/with_open.jl
using tempfile: NamedTemporaryFile
function main()
if true
temp_file = NamedTempFile::new()
file_path = name(temp_file)
if true
f = open(file_path, "w")
write(f, "hello");
end
if true
f = open(file_path, "r")
@assert(read(f, 1) == "h")
@assert(read(f) == "ello")
println("OK");
end
end
end

main()
